src/lib/datepicker/datepicker.ts

```diff
@@ -257,7 +257,7 @@ export class MdDatepicker<D> implements OnDestroy {
       this._ngZone.onStable.first().subscribe(() => this._popupRef.updatePosition());
     }
 
-    this._popupRef.backdropClick().first().subscribe(() => this.close());
+    this._popupRef.backdropClick().subscribe(() => this.close());
```

Comment:
Should already be fixed in master
https://github.com/angular/material2/commit/f89c6dbe905f0508810dda48fca55d4b41615764